index.js

```diff
@@ -38,32 +45,41 @@ module.exports = class MaxCache {
     this._gc()
   }
 
-  set (k, v) {
-    this._latest.set(k, v)
-    this._oldest.delete(k)
-    if (this._latest.size >= this.maxSize) this._gc()
+  set (k, v, options = {}) {
+    if (options.retain) this._retained.set(k, v)
```

Comment:
Should we make a special method that sets the retained value instead?

```
map.retained.set(k, v)
```

The `get` interop is the only important one

Reply:
I'm also pretty sure that setting a retained value won't behave correctly on `get()` if there was already a non-retained value in the cache. Let me just add a test.

Reply:
Perfect on the name

Reply:
If a key/value is retained it should prob be removed from the oldest/latest caches immediately

Reply:
And/or retained keys should always win i think

Reply:
@mafintosh Let me know what you think about ☝️